pkg/http/server_test.go

```diff
@@ -538,56 +542,6 @@ var _ = Describe("Server", func() {
 			cancel()
 		})
 
-		Context("with an ipv4 http server", func() {
```

Comment:
What's wrong with these test cases? 😛

Reply:
The bottom line here is that with this test suite, the coreos systemd library actually goes ahead and closes all the file descriptors it does not use. I made another version in another project that I can use instead, it skips the dependency and adds an override variable that ensures that the code does not close file descriptors that it should not close. Hopefully that will make it easier to implement the tests.

Reply:
I did figure it out, got some new code brewing.

Reply:
Ginkgo seems quite hostile when it comes to listeners, any hints how to please it? It worked well with this push it seems, but net after rebase.

Reply:
Not solved but a step in the right direction at least: https://github.com/oauth2-proxy/oauth2-proxy/pull/2539

Reply:
I'm hitting something like https://github.com/tailscale/tailscale/issues/7488
There's exactly one hit on the interwebs about it. It seems not nice to introduce it in the test suite.

Could we introduce the functionality without tests and let the testing become another PR?

Reply:
I did make some progress.
I was able to run just on spec and trigger a lot of problems.
Turns out that if I uncomment the TLS certificate generation, everything works, even though they are not used.
So I changed the library to another one (to generate cert), and now generation of certificates works, but not using them.

Not sure what's going on here really.

Reply:
Test are now fixed. I ended up running the binary instead, via gexec.